pkg/services/rendering/rendering.go

```diff
@@ -154,7 +155,7 @@ func (rs *RenderingService) Render(ctx context.Context, opts Opts) (*RenderResul
 }
 
 func (rs *RenderingService) render(ctx context.Context, opts Opts) (*RenderResult, error) {
-	if rs.inProgressCount > opts.ConcurrentLimit {
+	if int(atomic.LoadInt32(&rs.inProgressCount)) > opts.ConcurrentLimit {
```

Comment:
@marefr i am thinking could we just put concurrent_render_limit setting type to int64 or int32(if it is backward compatible) instead of cast forward/back int32/int.

Reply:
Yes I believe so